src/sqs/push.ts

```diff
@@ -39,3 +41,60 @@ export const pushQueueMessageHandler : MessageHandler<PushQueueMessagePayload> =
 	await processPush(github, payload, wrapLogger(context.log));
 }
 
+
+const NON_ERRONEOUS_STATUSES = [401, 404, 403];
+const RATE_LIMITING_DELAY_BUFFER = 10;
+const EXPONENTIAL_BACKOFF_BASE = 60;
+export const pushQueueErrorHandler : ErrorHandler<PushQueueMessagePayload> = async (error: JiraClientError | Octokit.HookError | RateLimitingError | Error,
+	context: Context<PushQueueMessagePayload>) : Promise<ErrorHandlingResult> => {
+
+	//Handle non-failure cases
+	const maybeResult = maybeHandleNonFailureCase(error, context);
+	if(maybeResult) {
+		return maybeResult;
+	}
+
+	//Handle failure cases
+	const errorHandlingResult = handleFailureCase(error, context);
+
+	if(!errorHandlingResult.retryable || context.lastAttempt ) {
+		//Emit metrics if the failure is not retryable
+		context.log.error({error}, "Webhook push processing failed and won't be retried anymore");
+		emitWebhookFailedMetrics("push")
+	}
+
+	return errorHandlingResult;
+}
+
+function maybeHandleNonFailureCase(error: Error, context: Context<PushQueueMessagePayload>): ErrorHandlingResult | undefined {
+	if(error instanceof JiraClientError &&
+		error.status &&
+		NON_ERRONEOUS_STATUSES.includes(error.status)) {
```

Comment:
nit: I'd suggest `NON_FAILURE_STATUSES`, because you've used word "failure" for this situation before (less notions => easier to follow IMHO)

Reply:
Actually, should it be something like `RETRYABLE_STATUSES`? I feel like "retryable/non-retryable" is more clear and explicit than "error/failure"....

Reply:
And the same goes to the names of functions (e.g. `maybeHandleRetryableCase`)...

Reply:
So the thing is when it is not a failure, we won't send a faulre metric. 

So I wanted to name it to reflect taht it is not a failure

Reply:
"Non-failure = we retry", is my understanding correct?